apps/src/applab/applab.js

```diff
@@ -1243,6 +1256,36 @@ function onInterfaceModeChange(mode) {
 }
 
 /**
+ * Handle a view change within data mode.
+ * @param {DataView} view
+ */
+function onDataViewChange(view, tableName) {
+  if (!studioApp.reduxStore.getState().pageConstants.hasDataMode) {
+    throw new Error('onDataViewChange triggered without data mode enabled');
+  }
+  const storageRef = getDatabase(Applab.channelId).child('storage');
+  switch (view) {
+    case DataView.OVERVIEW:
+      storageRef.off();
+      return;
+    case DataView.PROPERTIES:
+      storageRef.off();
+      storageRef.child('keys').on('value', snapshot => {
+        studioApp.reduxStore.dispatch(updateKeyValueData(snapshot.val()));
+      });
+      return;
+    case DataView.TABLE:
+      storageRef.off();
+      storageRef.child(`tables/${tableName}`).on('value', snapshot => {
+        studioApp.reduxStore.dispatch(updateTableData(snapshot.val()));
+      });
+      return;
+    default:
+      return;
+  }
+}
```

Comment:
I think the answer is no, because I'm showing/hiding these components rather than creating/destroying them. I'm doing that because I believe it will be better for performance reasons.